src/Symfony/Component/DependencyInjection/ServiceLocator.php

```diff
@@ -27,8 +27,9 @@
  * @template-covariant T of mixed
  *
  * @implements ServiceProviderInterface<T>
+ * @implements ServiceCollectionInterface<T>
  */
-class ServiceLocator implements ServiceProviderInterface, \Countable
+class ServiceLocator implements ServiceProviderInterface, ServiceCollectionInterface
```

Comment:
As said by @ro0NL, we already have interfaces to cover all the features.
```suggestion
class ServiceLocator implements ServiceProviderInterface, \Countable, \ IteratorAggregate
```